controller/GiUnplannedItem.controller.js

```diff
@@ -0,0 +1,803 @@
+				snTable.removeAllItems();
+				oModel.setProperty("/serialNumbers", []);
+				serialNumbers = [];
+				return;
+			}
+
+			// Remove extra serial number rows
+			while (qty < serialNumbers.length) {
+				serialNumbers.pop();
+			}
+
+			// Add additional rows
+			while (qty > serialNumbers.length) {
+				serialNumbers.push({
+					Sernr: ""
+				});
+			}
+
+			var colItemTemplate = sap.ui.getCore().byId(this.getView().createId("snColumnListItem"));
+			var tabModel = new sap.ui.model.json.JSONModel(serialNumbers, false);
+			tabModel.setData(serialNumbers, false);
+
+			var context = new sap.ui.model.Context(tabModel, "/");
+			snTable.setBindingContext(context);
+			snTable.setModel(tabModel);
+			snTable.bindItems("/", colItemTemplate);
+		},
+		filterTable: function(oTable, fields, filterString) {
+			// filter list
+			var oBinding = oTable.getBinding("items");
+			if (oBinding) {
+				var oFilters = [];
+				for (var i = 0; i < fields.length; i++) {
+					oFilters.push(new sap.ui.model.Filter(fields[i], sap.ui.model.FilterOperator.Contains, filterString));
+				}
+				var filterObj = new sap.ui.model.Filter(oFilters, false);
+				oBinding.filter(filterObj);
+			} else {
+				oBinding.filter([]);
+			}
+		},
+		liveChangeBin: function(oEvent) {
+			GlobalFuncs.uppercase(oEvent);
+			if (!oEvent.getSource().getValue()) {
+				return;
+			}
+			zController.getView().getModel().setProperty("/Matnr", "");
+			zController.getView().getModel().setProperty("/Verme", "");
+			zController.getView().getModel().setProperty("/Maktg", "");
+			zController.getView().getModel().setProperty("/Meins", "");
+			this.getView().getModel().setProperty("/Lgpla", oEvent.getSource().getValue());
+		},
+		changeBin: function(oEvent) {
+			GlobalFuncs.uppercase(oEvent);
+			zController.getView().getModel().setProperty("/Matnr", "");
+			zController.getView().getModel().setProperty("/Verme", "");
+			zController.getView().getModel().setProperty("/Maktg", "");
+			zController.getView().getModel().setProperty("/Meins", "");
+			this.getView().getModel().setProperty("/Lgpla", oEvent.getSource().getValue());
+			zController.f4Matnr();
+		},
+		f4Matnr: function() {
+			// Only show F4 if it is not already being displayed
+			// When user change bin and clicks Material field it would
+			// trigger F4 help twice.
+			if (sap.ui.getCore().byId(zController.getView().createId("f4MatnrTable"))){
+				sap.ui.getCore().byId(zController.getView().createId("f4MatnrTable")).destroy();
+			}
+			
+			sap.ui.core.BusyIndicator.show();
+
+			var oModel = this.getView().getModel();
+			oModel.setProperty("/Matnr", "");
+
+			if (!oModel.getProperty("/Lgpla")) {
+				GlobalFuncs.showMessage("Error", "Enter a Bin", "ERROR");
+				sap.ui.core.BusyIndicator.hide();
+				return;
+			}
+
+			var columnItems = new sap.m.ColumnListItem({
+				type: "Active" //enables itemPress event
+			});
+			//create cells to add to the rows.
+			var Matnr = new sap.m.Text({

```

Comment:
same here, why are we creating all the controls in JS while we can use the XML view as a template? I see this in almost all the controls... it will be nice if we can use more the XML views instead having so many JS complexity